e2e/app/create-account-page.ts

```diff
@@ -1,187 +1,206 @@
+    const dropdown = new SelectComponent(this.page, FIELD_LABEL.EDUCATION_LEVEL);
     await dropdown.select(selectTextValue);
   }
 
   // select Year of Birth from a dropdown
-  public async selectYearOfBirth(year: string) {
-    const dropdown = new DropdownSelect(this.puppeteerPage, 'Year of Birth');
+  async selectYearOfBirth(year: string) {
+    const dropdown = new SelectComponent(this.page, FIELD_LABEL.YEAR_OF_BIRTH);
     await dropdown.select(year);
   }
 
   // Combined steps to make test code cleaner and shorter
 
   // Step 1: Enter Invitation key
-  public async fillOutInvitationKey(invitationKey: string) {
+  async fillOutInvitationKey(invitationKey: string) {
     await this.getInvitationKeyInput()
     .then(invitationKeyInput => invitationKeyInput.type(invitationKey))
     .then(() => this.getNextButton())
     .then(submitButton => submitButton.click());
   }
 
   // Step 2: Accepting Terms of Use and Privacy statement.
-  public async acceptTermsOfUseAgreement() {
-    const privacyStatementCheckbox = await this.getPrivacyStatementCheckbox();
-    const termsOfUseCheckbox = await this.getTermsOfUseCheckbox();
-    const nextButton = await this.getNextButton();
+  async acceptTermsOfUseAgreement() {
+    await this.getPrivacyStatementCheckbox();
+    await this.getTermsOfUseCheckbox();
+    await this.getNextButton();
 
     await this.scrollToLastPdfPage();
 
     // check by click on label works
-    await (await this.getPrivacyStatementLabel()).click();
-    await (await this.getTermsOfUseLabel()).click();
-
-    // TODO uncomment after bug fixed https://precisionmedicineinitiative.atlassian.net/browse/RW-4487
-    // uncheck a checkbox to ensure NEXT button becomes disabled
-    // await page.evaluate(e => e.click(), await createAccountPage.getTermsOfUseLabel());
-    // back to check on
-    // await page.evaluate(e => e.click(), await createAccountPage.getTermsOfUseLabel());
+    await (await this.getPrivacyStatementCheckbox()).check();
+    await (await this.getTermsOfUseCheckbox()).check();
   }
 
   // Step 3: Enter user default information
-  public async fillOutUserInformation() {
-    const newUserName = await this.fillInFormFields(registrationFields.inputFieldsValues);
+  async fillOutUserInformation() {
+    const newUserName = await this.fillInFormFields(defaultFieldValues);
     await (await this.getResearchBackgroundTextarea()).type(faker.lorem.word());
+    // a different dropdown selection depending on Yes or No radiobutton was selected
+    await (await this.areYouAffiliatedRadioButton(true)).click();
     await (await this.getInstitutionNameInput()).type(faker.company.companyName());
-    await this.selectInstitution(registrationFields.institutionAffiliation.EARLY_CAREER_TENURE_TRACK_RESEARCHER);
-    await this.puppeteerPage.waitFor(1000);
+    await this.selectInstitution(INSTITUTION_AFFILIATION.EARLY_CAREER_TENURE_TRACK_RESEARCHER);
+    await this.page.waitFor(1000);
```

Comment:
Per my previous comment, I would really like that we eliminate any arbitrary sleeps in the tests, or else have clear comments about why they are absolutely necessary.

Reply:
1 second pause is needed, else test becomes flaky because Puppeteer is very fast. I added a comment.
`// need pause 1 second for dropdown to disappear, so it is not blocking click on elements below.`

Reply:
For example, in this case - could we instead block until the dropdown element is no longer visible, rather than sleeping?